pkg/operator/hive/hiveadmission.go

```diff
@@ -120,12 +114,13 @@ func (r *ReconcileHiveConfig) deployHiveAdmission(hLog log.FieldLogger, h resour
 	asset := assets.MustAsset(deploymentAsset)
 	hLog.Debug("reading deployment")
 	hiveAdmDeployment := resourceread.ReadDeploymentV1OrDie(asset)
+	hiveAdmContainer := &hiveAdmDeployment.Spec.Template.Spec.Containers[0]
```

Comment:
Isn't this the spot where we fetch-the-hive-container-by-name-out-of-the-deployment step goes? Otherwise we're still just hard-coding an assumption about the way the hive Deployment resource looks like.

Reply:
Well, that's how the code was working before. Also this follows the pattern in the other file. This (and the corresponding) function is inextricably tied to what's in the deployment manifest. The functions would have to be reconsidered wholesale if we made significant changes to the shape of the manifests. I can understand moving the assumption of `[0]` away from the fringe helper functions and into this primary one; but adding logic here that looks for "the container named foo" as opposed to "the first and only container" seems like overkill.

Maybe I'm missing something. Let's chat.

Reply:
[Agreed](https://coreos.slack.com/archives/CE3ETN3J8/p1639676515068100?thread_ts=1639676162.066200&cid=CE3ETN3J8) to do this in a [followon](https://github.com/openshift/hive/pull/1643).